tests/tck-build-logic/src/main/groovy/org.graalvm.internal.tck-harness.gradle

```diff
@@ -102,6 +103,60 @@ def matrixDefault = [
 
 final String METADATA_GROUP = "Metadata"
 
+List<String> getNewerVersionsFor(String library, String startingVersion) {
+    def baseUrl = "https://repo1.maven.org/maven2"
+    String[] libraryParts = library.split(":")
+    String group = libraryParts[0].replace(".", "/")
+    String artifact = libraryParts[1]
+    def data = new URL(baseUrl + "/" + group + "/" + artifact + "/" + "maven-metadata.xml").getText()
+
+    return tck.getNewerVersionsFromLibraryIndex(data, startingVersion, library)
+}
+
+// gradle getExistingLibrariesWithNewVersions
+Provider<Task> getExistingLibrariesWithNewVersions = tasks.register("getExistingLibrariesWithNewVersions", DefaultTask) { task ->
```

Comment:
:x: Would be cleaner to move this task code to a proper task in the `tck-build-logic`. There's a lot of build logic in this and it's not a good practice to mix build logic with scripts.

Reply:
nit: the name `getExistingLibrariesWithNewVersions` is not very Gradle-idiomatic, it sounds like a getter rather than a task name. maybe `fetch` instead of `get`?